File: GeeksForGeeks/Easy/Encrypt_the_string_2.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main()
{
    int test_case;
    cin >> test_case;
    string str, result;
    char test_character;
    int test_count;
    while (test_case--)
    {
        cin >> str;
        test_character = str[0];
        test_count = 1;
        result = "";
        char a[50];
        for(int i = 1; i < str.length(); ++i)
        {
            if (test_character == str[i])
            {
                ++test_count;
            }
            else
            {
                sprintf(a, "%x", test_count);
                result += test_character;
                result += a;
                test_character = str[i];
                test_count = 1;
            }
        }
        result += test_character;
        sprintf(a, "%x", test_count);
        result += a;
        reverse(result.begin(), result.end());
        cout << result << endl;
    }
}

File: GeeksForGeeks/Easy/Encrypt_the_string_2.java
import java.util.Scanner;

class GFG
 {

  public static String reverse(String input)
  {
      String output = "";
      for(int i = 0; i < input.length(); ++i)
      {
          output = input.charAt(i) + output;
      }
      return output;
  }
	public static void main (String[] args)
	{
	    Scanner scan = new Scanner(System.in);
	    int testcase = scan.nextInt();
      String str, result;
      char testCharacter;
      int testCount;
      scan.nextLine();
    	while(testcase!=0)
    	{
    	    str = scan.nextLine();
          testCharacter = str.charAt(0);
          testCount = 1;
          result = "";
          for(int i = 1; i < str.length(); ++i)
          {
              if (testCharacter == str.charAt(i))
              {
                  ++testCount;
              }
              else
              {
                  result += testCharacter + "" + Integer.toHexString(testCount);
                  testCharacter = str.charAt(i);
                  testCount = 1;
              }
          }
          result += testCharacter + "" + Integer.toHexString(testCount);

    	    System.out.println(reverse(result));
          --testcase;
    	}
    	scan.close();
	}
}
